usecases/LDpred2/ldpred2.R

```diff
@@ -153,47 +209,77 @@ for (chr in chromosomes) {
       corr$add_columns(corr0, nrow(corr))
     }
   } else {
-    cat('Skipped chromosome ', chr, '. Reason: Nr of SNPs was ', nMarkers, '\n')
+    cat('\nSkipped chromosome ', chr, '. Reason: Nr of SNPs was ', nMarkers, '\n', sep='')
     df_beta <- df_beta[df_beta$chr != chr,]
   }
 }
+nrMissingLDs <- sum(is.na(ld))
+if (nrMissingLDs > 0) {
+  details <- ifelse(genoLDSample > 0, paste0(genoLDSample, ' individuals used, which may be too small'))
+  stop('Missing LD blocks (', nrMissingLDs,')! ', details)
+}
 
-cat('Running LD score regression\n')
+cat('\n### Running LD score regression\n')
 ldsc <- with(df_beta, snp_ldsc(ld, length(ld), chi2=(beta/beta_se)^2, sample_size=n_eff, blocks=NULL))
 h2_est <- ldsc[["h2"]]
-cat('SNP heritability estimated to ', h2_est, '\n')
+cat('Results:', 'Intercept =', ldsc[["int"]], 'H2 =', h2_est, '\n')
 
+cat('\n### Starting polygenic scoring\n')
 # LDPRED2-Inf: Infinitesimal model
 if (argLdpredMode == 'inf') {
   cat ('Running LDPRED2 infinitesimal model\n')
   cat('Calculating beta inf\n')
-  beta_inf <- snp_ldpred2_inf(corr, df_beta, h2=h2_est)
-  cat('Calculating PGS\n')
-  pred_inf <- big_prodVec(G, beta_inf, ind.row=ind.test, ind.col=df_beta[['_NUM_ID_']])
-
-  #print(pcor(pred_inf, y[ind.test], NULL))
-  cat('Calculating PGS for all individuals\n')
-  pred_all <- big_prodVec(G, beta_inf, ind.col=df_beta[['_NUM_ID_']])
-  obj.bigSNP$fam$score <- pred_all
+  beta <- snp_ldpred2_inf(corr, df_beta, h2=h2_est)
 # LDPRED2-Auto
 } else if (argLdpredMode == 'auto') {
-  cat ('Running LDPRED2 auto model\n')
-  multi_auto <- snp_ldpred2_auto(corr, df_beta, h2_init=h2_est, vec_p_init=seq_log(1e-4,0.2,length.out=5), allow_jump_sign=F, 
-                                 shrink_corr=0.95, ncores=NCORES)
-  cat('Evaluating\n')
+  cat('Running LDPRED2 auto model\n')
+  if (!is.na(setSeed)) set.seed(setSeed)
+  multi_auto <- snp_ldpred2_auto(corr, df_beta, h2_init=h2_est, vec_p_init=seq_log(1e-4, 0.2, length.out=parHyperPLength), 
+                                 allow_jump_sign=F, shrink_corr=0.95, ncores=NCORES)
+  cat('Plotting diagnostics: ', fileOutput, '.png\n', sep='')
+  library(ggplot2)
+  auto <- multi_auto[[1]]
+  plt <- plot_grid(
+    qplot(y = auto$path_p_est) + theme_bigstatsr() +
```

Comment:
This qplot function in qqplot2 3.4.0 is deprecated, see https://ggplot2.tidyverse.org/reference/qplot.html). A warning is thrown, so not an immediate concern. I see the version I specified to be installed in the `r.sif` build is 3.3.5, however, 3.4.0 is installed for some reason.

Reply:
Opened issue #104 for this.

Reply:
I can rewrite the plotting code using pure ggplot, should be quite simple

Reply:
> I can rewrite the plotting code using standard ggplot, should be quite simple

Ok, that'd be more future proof

Reply:
I'm marking this as resolved if it is blocking merging, as it is not a high priority